src/map/lua/lua_baseentity.cpp

```diff
@@ -12666,28 +12666,14 @@ bool CLuaBaseEntity::delLatent(uint16 condID, uint16 conditionValue, uint16 mID,
 
 int16 CLuaBaseEntity::getMaxGearMod(Mod modId)
 {
-    CCharEntity* PChar       = dynamic_cast<CCharEntity*>(m_PBaseEntity);
-    uint16       maxModValue = 0;
-
-    if (!PChar)
+    if (m_PBaseEntity->objtype != TYPE_PC)
     {
-        ShowWarning("CLuaBaseEntity::getMaxGearMod() - m_PBaseEntity is not a player.");
+        ShowWarning("Invalid Entity (NPC: %s) calling function.", m_PBaseEntity->GetName());
```

Comment:
This could get called on a Pet or a Mob, not necessarily just an NPC

Reply:
Changed it to Non-PC, since it is meant for player characters only.